### PR TITLE
created required end points

### PR DESCRIPTION
**Pull request by devmarye**
Created the endpoints required

### CODE REVIEW

File: server/controllers/MentorController.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+        } catch (error) {
+            res.status(500).json({error:error.message})
+        }
+    },
+
+    async createMentor(req, res) {
+        try {
+           const hashedPassword = await bcrypt.hash(req.body.password, 10);
+           const newMentor = await pool.query("INSERT INTO mentors ( firstName, lastName, email,password,address,bio,occupation,expertise) VALUES ($1, $2,$3,$4,$5,$6,$7,$8) RETURNING *", [req.body.firstName, req.body.lastName, req.body.email, hashedPassword, req.body.address, req.body.bio, req.body.occupation, req.body.expertise])
```
Comment (TheoOkafor) [inline server/controllers/MentorController.js:21]: @brendagold use object destructuring here.
Eg.
```
const { firstName, lastName, ...} = req.body;
```
Your code will look cleaner that way

File: server/controllers/MentorController.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        } catch (error) {
+            res.status(500).json({error:error.message})
+        }
+    },
+
+    async createMentor(req, res) {
+        try {
+           const hashedPassword = await bcrypt.hash(req.body.password, 10);
+           const newMentor = await pool.query("INSERT INTO mentors ( firstName, lastName, email,password,address,bio,occupation,expertise) VALUES ($1, $2,$3,$4,$5,$6,$7,$8) RETURNING *", [req.body.firstName, req.body.lastName, req.body.email, hashedPassword, req.body.address, req.body.bio, req.body.occupation, req.body.expertise]) 
+           res.json({mentor: newMentor.rows[0]})
```
Comment (TheoOkafor) [inline server/controllers/MentorController.js]: Follow the specifications given in the document to do this. Check the specification for the payload

Comment (TheoOkafor) [reply server/controllers/MentorController.js]: Create a responseFormat helper function that will help you to keep your response in a standard format based on the specification

File: server/controllers/MentorController.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+
+
+    async getMentorById(req, res) {
+        try {
+          const id = req.params.mentorid;
+          const mentor = await pool.query("Select * From mentors WHERE mentorid = $1", [
+            id,
+          ]);
+          res.status(200).json({ mentor: mentor.rows[0] });
```
Comment (TheoOkafor) [inline server/controllers/MentorController.js]: What response is returned when the mentor do not exist ie. when the mentor is not found?

File: server/controllers/SessionController.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    } catch (error) {
+      res.status(500).json({ error: error.message });
+    }
+  },
+
+  async createSession(req, res) {
+    try {
+      const newUser = await pool.query(
+        "INSERT INTO sessions ( mentorid, questions, menteeid, menteeEmail) VALUES ($1, $2,$3,$4) RETURNING *",
+        [req.body.mentorid, req.body.questions, req.user.userid, req.user.email]
```
Comment (TheoOkafor) [inline server/controllers/SessionController.js]: Don't forget destructuring

File: server/controllers/SessionController.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  async acceptSession(req, res) {
+    try {
+      const sessionid = req.params.sessionid;
+      const update  = "accepted";
+      const response = await pool.query(
+        "UPDATE sessions SET status = $1 WHERE sessionid = $2 RETURNING *",
+        [update, sessionid]
+      );
+
+      res.json({ data: response.rows[0]})
```
Comment (TheoOkafor) [inline server/controllers/SessionController.js]: What happens when the session is not found or invalid

File: server/controllers/SessionController.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  async rejectSession(req, res) {
+    try {
+        const sessionid = req.params.sessionid;
+        const update  = "rejected";
+        const response = await pool.query(
+          "UPDATE sessions SET status = $1 WHERE sessionid = $2 RETURNING *",
+          [update, sessionid]
+        );
+  
+        res.json({ data: response.rows[0]})
```
Comment (TheoOkafor) [inline server/controllers/SessionController.js]: Same here. What if the session with the id do not exist.

File: server/controllers/UserController.js [hunk 7]
```diff
@@ -1,27 +1,84 @@
...
 
 export default {
-    async allUsers(req,res) {
-        try {
-           const users = await pool.query('SELECT * FROM users');
-           res.json({users: users.rows})
+  async allUsers(req, res) {
+    try {
+      const users = await pool.query("SELECT * FROM users");
+      res.status(200).json({ users: users.rows });
```
Comment (TheoOkafor) [inline server/controllers/UserController.js]: Does this not return with the passwords of all the users in the database? Will you like it if facebook had an endpoint that returns your password?

File: server/controllers/UserController.js [hunk 8]
```diff
@@ -1,27 +1,84 @@
...
-            res.status(500).json({error:error.message})
-        }
-    },
+  async getUserById(req, res) {
+    try {
+      const id = req.params.userid;
+      const user = await pool.query("Select * From users WHERE userid = $1", [
+        id,
+      ]);
+      res.status(200).json({ user: user.rows });
```
Comment (TheoOkafor) [inline server/controllers/UserController.js]: Does this not return the user password?

File: server/db.js [hunk 9]
```diff
@@ -2,12 +2,14 @@ import pg from "pg";
...
 
+
+
 let localPoolConfig = {
   user: "postgres",
   password: "Brenda@919",
   host: "localhost",
   port: "5432",
-  database: "freementors",
+  database: "postgres",
```
Comment (TheoOkafor) [inline server/db.js]: Put these in an environment variable.

**Review by TheoOkafor — CHANGES_REQUESTED** [review]
Nice effort. You are unto something. 

Attend to the comments please.

File: server/controllers/MentorController.js (hunk 1 above)
Comment (devmarye) [reply server/controllers/MentorController.js:21]: Okay

File: server/controllers/UserController.js (hunk 7 above)
Comment (devmarye) [reply server/controllers/UserController.js]: Hahahaha, no oo

File: server/db.js (hunk 9 above)
Comment (devmarye) [reply server/db.js]: I am not able to put the user and password in an ENV

I get these errors:
The db user changed to my PC user
while for Password: SASL: SCRAM-SERVER-FIRST-MESSAGE: client password must be a string

Comment (TheoOkafor) [reply server/db.js]: have you tried googling the error?

File: server/controllers/MentorController.js (hunk 1 above)
Comment (TheoOkafor) [reply server/controllers/MentorController.js:21]: I don't see destructuring here

File: server/db.js [hunk 10]
```diff
@@ -2,12 +2,14 @@ import pg from "pg";
 
 const { Pool } = pg;
 
+
+
 let localPoolConfig = {
   user: "postgres",
   password: "Brenda@919",
```
Comment (TheoOkafor) [inline server/db.js:9]: still find a way to put these in the env

**Review by TheoOkafor — CHANGES_REQUESTED** [review]

**Pull request merged**